.github/workflows/ci.yml

```diff
@@ -61,7 +61,6 @@ jobs:
       fail-fast: false
       matrix:
         python-version: [
-          "3.7",
```

Comment:
I'd like to do this in a separate pull request, as discussed offline, since it's going to many small changes that are best reviewed separately.

Reply:
Yes, I'll create a separate PR. I removed it here temporarily to stop seeting the 3.7 failures.